src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +583,105 @@ Namespace Microsoft.CodeAnalysis.Semantics
+            Dim conversionInformation = CreateConversionOperand(boundConversion.Operand, boundConversion.ConversionKind, boundConversion.Syntax, boundConversion.Type)
+            Dim methodSymbol As MethodSymbol = conversionInformation.methodSymbol
+            Dim operand As Lazy(Of IOperation) = conversionInformation.operation
+            Dim isDelegateCreation As Boolean = conversionInformation.isDelegateCreation
+
+            If isDelegateCreation Then
+                ' If this is a conversion from a lambda to a delegate type, or this is an AddressOf delegate creation
+                ' as determined above, we return a DelegateCreationExpression, instead of returning a conversion expression
+                Return New LazyDelegateCreationExpression(operand, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
 
-            Dim conversion = New Conversion(New KeyValuePair(Of ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
+            Dim conversion = New Conversion(New KeyValuePair(Of VisualBasic.ConversionKind, MethodSymbol)(boundConversion.ConversionKind, methodSymbol))
             Dim isExplicit As Boolean = boundConversion.ExplicitCastInCode
             Dim isTryCast As Boolean = False
             Dim isChecked As Boolean = False
-            Dim type As ITypeSymbol = boundConversion.Type
-            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundConversion.ConstantValueOpt)
-            Dim isImplicit As Boolean = boundConversion.WasCompilerGenerated
             Return New LazyVisualBasicConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
+        Private Shared Function IsDelegateCreation(conversionKind As ConversionKind, conversionSyntax As SyntaxNode, operand As BoundNode, targetType As TypeSymbol) As Boolean
+            ' Any of the explicit cast types, as well as New DelegateType(AddressOf Method)
+            ' Additionally, AddressOf, if the child AddressOf is the same SyntaxNode (ie, an implicit delegate creation)
+            Dim validAddressOfConversionSyntax = (conversionSyntax.Kind() = SyntaxKind.CTypeExpression OrElse
+                                                  conversionSyntax.Kind() = SyntaxKind.DirectCastExpression OrElse
+                                                  conversionSyntax.Kind() = SyntaxKind.TryCastExpression OrElse
+                                                  conversionSyntax.Kind() = SyntaxKind.ObjectCreationExpression OrElse
+                                                  (conversionSyntax.Kind() = SyntaxKind.AddressOfExpression AndAlso
+                                                   ReferenceEquals(conversionSyntax, operand.Syntax))) AndAlso
+                                                 operand.Syntax.Kind() = SyntaxKind.AddressOfExpression
```

Comment:
>operand.Syntax.Kind() = SyntaxKind.AddressOfExpression [](start = 49, length = 54)

I think it would be good to strengthen this condition with ```operand.Kind = BoundKind.DelegateCreationExpression```, or replace it with this condition.

Reply:
Unfortunately, the operand there isn't always a BoundDelegateCreationExpression. In my tests, it can also be a BoundAddressOfOperator or a BoundBadExpression, which is why I just went with a syntax check
I can swap that out for checking each of the 3 operand kinds, but it seemed simpler to me to just have a syntax check.